integration_tests/tests/transactionCSV-to-s3-event-tests.ts

```diff
@@ -1,97 +1,70 @@
-import { waitForTrue } from "../../src/handlers/int-test-support/helpers/commonHelpers";
+import { ListObjectsCommandOutput } from "@aws-sdk/client-s3";
+import { poll } from "../../src/handlers/int-test-support/helpers/commonHelpers";
 import { resourcePrefix } from "../../src/handlers/int-test-support/helpers/envHelper";
-import { addFauxDataToTestCases } from "../../src/handlers/int-test-support/helpers/mock-data/csv/addFauxDataToTestCases";
-import { objectsToCSV } from "../../src/handlers/int-test-support/helpers/mock-data/csv/objectsToCsv";
-import { testCases } from "../../src/handlers/int-test-support/helpers/mock-data/csv/transformCSV-to-event-test-data";
+import { mockIdpCsvData } from "../../src/handlers/int-test-support/helpers/mock-data/csv";
 import {
   deleteS3Objects,
   getS3Object,
   listS3Objects,
   putS3Object,
-  S3Object,
 } from "../../src/handlers/int-test-support/helpers/s3Helper";
 
-const prefix = resourcePrefix();
-const folderPrefix = "btm_transactions";
-const bucketName = `${prefix}-storage`;
+const transactionsDirectory = "btm_transactions";
+const outputBucket = `${resourcePrefix()}-storage`;
+const inputBucket = `${resourcePrefix()}-transaction-csv`;
 
-const checkS3BucketForEventIds = async (): Promise<boolean> => {
-  const result = await listS3Objects({
-    bucketName,
-    prefix: folderPrefix,
-  });
-  if (
-    result.Contents === undefined ||
-    result.Contents.length !==
-      testCases.filter((data) => data.expectedPath === "happy").length
-  ) {
-    return false;
-  } else {
-    return true;
-  }
-};
+const { testCases, csv } = mockIdpCsvData();
 
-describe("\n Given a csv with event data is uploaded to the transaction csv bucket", () => {
+describe("Given a csv with event data is uploaded to the transaction csv bucket", () => {
   beforeAll(async () => {
     // First delete all transactions in btm_transaction
-    await deleteS3Objects({ bucketName, prefix: folderPrefix });
-    // Add data to build the test csv
-    const augmentedData = addFauxDataToTestCases(testCases, {
-      Timestamp: "2023-01-01T00:27:41.186Z",
-      "RP Entity Id": "fake rp entity id",
+    await deleteS3Objects({
+      bucketName: outputBucket,
+      prefix: transactionsDirectory,
     });
-    // Filter and rename data in the test csv
-    const csvString = objectsToCSV(augmentedData, {
-      filterKeys: ["expectedPath"],
-      renameKeys: new Map([
-        ["givenEntity", "Idp Entity Id"],
-        ["givenLoa", "Minimum Level Of Assurance"],
-        ["givenStatus", "Billable Status"],
-        ["expectedEventId", "Request Id"],
-      ]),
-    });
```

Comment:
`addFauxDataToTestCases` (renamed to `augmentTestCases`) and `objectsToCSV` moved into an orchestration function `mockIdpCsvData` to reduce the length of this `beforeAll`